multicluster/cmd/multicluster-controller/clusterset_webhook.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	mcv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+)
+
+//+kubebuilder:webhook:path=/validate-multicluster-crd-antrea-io-v1alpha1-clusterset,mutating=false,failurePolicy=fail,sideEffects=None,groups=multicluster.crd.antrea.io,resources=clustersets,verbs=create;update,versions=v1alpha1,name=vclusterset.kb.io,admissionReviewVersions={v1,v1beta1}
+
+// ClusterSet validator
+type clusterSetValidator struct {
+	Client    client.Client
+	decoder   *admission.Decoder
+	namespace string
+}
+
+// Handle handles admission requests.
+func (v *clusterSetValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	clusterSet := &mcv1alpha1.ClusterSet{}
+	err := v.decoder.Decode(req, clusterSet)
+	if err != nil {
+		klog.ErrorS(err, "Error while decoding ClusterSet", "ClusterSet", req.Namespace+"/"+req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	oldClusterSet := &mcv1alpha1.ClusterSet{}
+	if req.OldObject.Raw != nil {
+		if err := json.Unmarshal(req.OldObject.Raw, &oldClusterSet); err != nil {
+			klog.ErrorS(err, "Error while decoding old ClusterSet", "ClusterSet", klog.KObj(clusterSet))
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+		if oldClusterSet.Spec.Leaders[0].ClusterID != clusterSet.Spec.Leaders[0].ClusterID {
+			klog.ErrorS(err, "the field 'clusterID' is immutable", "ClusterSet", klog.KObj(clusterSet))
+			return admission.Denied("the field 'clusterID' is immutable")
+		}
+		return admission.Allowed("")
+	}
+
+	// Check if there is any existing ClusterSet.
+	clusterSetList := &mcv1alpha1.ClusterSetList{}
+	if err := v.Client.List(context.TODO(), clusterSetList, client.InNamespace(v.namespace)); err != nil {
+		klog.ErrorS(err, "Error reading ClusterSet", "Namespace", v.namespace)
+		return admission.Errored(http.StatusPreconditionFailed, err)
+	}
+
+	if len(clusterSetList.Items) > 0 {
+		klog.ErrorS(fmt.Errorf("multiple ClusterSets in the Namespace are not allowed"),
```

Comment:
Why not create a single error to be used both here and in the returned error?

Reply:
"multiple ClusterSets in a Namespace are not allowed" or "only a single ClusterSet is allowed in a Namespace"

Reply:
Done. defined an error `"multiple ClusterSets in a Namespace are not allowed"`